browser/registry.go

```diff
@@ -69,10 +85,44 @@ func newRemoteRegistry(envLookup env.LookupFunc) *remoteRegistry {
 		parts = parts[:len(parts)-1]
 	}
 
-	r.isRemote = true
-	r.wsURLs = parts
+	return true, parts
+}
+
+// checkForScenarios will parse the K6_INSTANCE_SCENARIOS env var if
+// it has been defined.
+func checkForScenarios(envLookup env.LookupFunc) (bool, []string, error) {
+	scenariosJSON, isRemote := envLookup("K6_INSTANCE_SCENARIOS")
+	if !isRemote {
+		return false, nil, nil
+	}
+
+	var scenarios []struct {
+		ID       string `json:"id"`
+		Browsers []struct {
+			Handle string `json:"handle"`
+		} `json:"browsers"`
+	}
+
+	err := json.Unmarshal([]byte(scenariosJSON), &scenarios)
+	if err != nil {
+		return false, nil, fmt.Errorf("parsing K6_INSTANCE_SCENARIOS: %w", err)
+	}
+
+	var wsURLs []string
+	for _, s := range scenarios {
+		for _, b := range s.Browsers {
+			if b.Handle == "" {
```

Comment:
To be on the safe side?

```suggestion
			if strings.TrimSpace(b.Handle) == "" {
```